### DISCUSSION TITLE
Would it be possible to enable kernel mode setting by default?

### DISCUSSION THREAD

**Discussion by ghost** [NVIDIA/open-gpu-kernel-modules#481] [Ideas]
Kernel mode setting is already enabled by default in most packaged Nvidia drivers and it's been around for a while so we might as well enable it by default now

**Comment by aritger** [comment]
Thanks for asking.

Yes, we're working towards that goal.  The challenge thus far has been that there are a few specialty features, most notably SLI Mosaic, that define special state in the NVIDIA driver stack, and need to define that state before any other NVIDIA driver component initializes things.

nvidia-drm.modeset=1 causes the NVIDIA driver components to be initialized before the SLI Mosaic state can be set.

There are of course many short-cut ways to solve that, but we've been trying to refactor things such that SLI Mosaic doesn't require early, driver-wide, special state.  That refactoring is just about done, finally.  After that, we can make nvidia-drm.modeset=1 the default.

**Comment by ghost** [comment]
Off topic but something I'm curious about. Is there any plans to provide official packages for some distros in addition to the run file?

**Reply by aritger** [reply]
We definitely recommend using distro-specific packages, rather than the .run file, when possible.  It is great when distros or other third parties repackage the driver for a particular distro.  I know CUDA releases will repackage the driver for some distros, but I'm not aware of any plans currently for NVIDIA to do that more generally.
